### PR TITLE
User can save assessment CANS-72

### PR DESCRIPTION
**Pull request by jonbwalker**
## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [ ] My code adds no new issues to Code Climate
- [ ] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: client/src/components/Assessment/AssessmentFormHeader.test.js [hunk 1]
```diff
@@ -11,11 +12,13 @@ describe('<AssessmentFormHeader />', () => {
       const wrapper = shallow(<AssessmentFormHeader {...props} />);
       const selector = '#child-name';
 
-      expect(wrapper.find(selector).text()).toBe('test child');
+      expect(wrapper.find(selector).text()).toBe('child, test');
+      expect(wrapper.find(selector).text()).not.toBe('child test');
```
Comment (denys-davydov) [inline client/src/components/Assessment/AssessmentFormHeader.test.js]: Not sure we need 3 'not.toBe' assertions here, they look redundant to me

Comment (jonbwalker) [reply client/src/components/Assessment/AssessmentFormHeader.test.js]: Fixed! ef46340

File: client/src/components/Assessment/Assessment.test.js [hunk 2]
```diff
@@ -1,38 +1,339 @@
...
+    });
+
+    describe('assessment with an existing assessment', () => {
+
+      it('calls fetchAssessment', () => {
+        const props = {
+          location: { childId: 1 },
+          match: { params: { id: 1 } },
+        };
+        const fetchAssessmentSpy = jest.spyOn(
```
Comment (edgibbs) [inline client/src/components/Assessment/Assessment.test.js]: Same thing with mocking the class under test.

File: client/src/components/Assessment/Assessment.test.js [hunk 3]
```diff
@@ -1,38 +1,339 @@
...
+
+        assessmentServicePostSpy.mockClear();
+      });
+
+      it('should call initialSave', async () => {
+        const props = {
+          location: { childId: 1 },
+          match: { params: { id: 1 } },
+        };
+        const initialSaveSpy = jest.spyOn(
```
Comment (edgibbs) [inline client/src/components/Assessment/Assessment.test.js]: Same idea here with mocking.

File: client/src/components/Assessment/Assessment.test.js [hunk 4]
```diff
@@ -1,38 +1,339 @@
...
 
-  describe('with an existing assessment', () => {
-    it('calls fetchAssessment', () => {
+  describe('initial save', () => {
+    it('will update the assessment on the component state', () => {
       const props = {
-        location: { clientFirstName: 'test', clientLastName: 'child' },
+        location: { childId: 1 },
         match: { params: { id: 1 } },
+        history: { push: jest.fn() },
```
Comment (edgibbs) [inline client/src/components/Assessment/Assessment.test.js]: I found the whole history concept confusing, how is the caller supposed to know they have to set a history within the props for this to do an initial save?

File: client/src/components/Assessment/AssessmentFormHeader.test.js (hunk 1 above)
Comment (edgibbs) [inline client/src/components/Assessment/AssessmentFormHeader.test.js]: If the first assertion passes, these others will always pass as well, so not sure they add anything.

File: client/src/components/Assessment/AssessmentFormHeader.test.js [hunk 5]
```diff
@@ -25,4 +28,67 @@ describe('<AssessmentFormHeader />', () => {
...
+
+  describe('select completedAs', () => {
+    it('should default Communimetric', () => {
+      const value = 'Communimetric';
+
+      const wrapper = shallow(<AssessmentFormHeader assessmentCompletedAs={value}/>);
+      expect(wrapper.instance().props.assessmentCompletedAs).toBe(value);
+    });
+
+    it('should allow user to completedAs', () => {
```
Comment (edgibbs) [inline client/src/components/Assessment/AssessmentFormHeader.test.js]: Should it be 'should allow user to **set** completedAs'

File: client/src/components/Assessment/AssessmentFormHeader.test.js [hunk 6]
```diff
@@ -25,4 +28,67 @@ describe('<AssessmentFormHeader />', () => {
...
+
+      expect(wrapper.instance().props.assessmentDate).toBe(today);
+      wrapper.instance().handleDateChange(tomorrow);
+      expect(handleChange).toBeCalled();
+      expect(handleChange).toBeCalledWith(tomorrow.target.value);
+    });
+  });
+
+  describe('select completedAs', () => {
+    it('should default Communimetric', () => {
```
Comment (edgibbs) [inline client/src/components/Assessment/AssessmentFormHeader.test.js]: OK, pretty much grammer at this point, but 'should default **to** Communimetric'

File: client/src/components/Assessment/Assessment.test.js (hunk 4 above)
Comment (edgibbs) [reply client/src/components/Assessment/Assessment.test.js]: make sense n

**Comment by edgibbs** [comment]
retest this please

**Comment by edgibbs** [comment]
retest this please

**Comment by edgibbs** [comment]
Minor codeclimate issues are dups in tests which are fine.

**Pull request merged**